### PR TITLE
Prep for semver bump and document current blocker 

### PR DESCRIPTION
**Pull request by jtgeibel**
This partially reverts the revert (#3051) of #2990. While we can't bump the version yet, it is worth recovering these changes and documenting the remaining work.

r? @Turbo87

### CODE REVIEW

**Comment by Turbo87** [comment]
looks like CI is unhappy 😢

**Comment by jtgeibel** [comment]
Forgetting to `cargo fmt` strikes again. Should work now.

**Comment by Turbo87** [comment]
you linked https://github.com/rust-lang/crates.io/pull/3094 in the description, but I'm assuming that was a typo, right? 🤔

**Comment by Turbo87** [comment]
I believe https://github.com/steveklabnik/semver/issues/217 might still be blocking this

**Comment by jtgeibel** [comment]
@Turbo87 you're right, I meant to link to #3049. And you're right, I think the linked issue is blocking this.

**Comment by jtgeibel** [comment]
@Turbo87 I've rolled back the version bump and added a comment pointing to the blocker. I've updated the PR title and description to reflect the new scope.

**Review by Turbo87 — APPROVED** [review]
would it make sense for us to write a failing test for the blocker so that we don't update accidentally?

feel free to r=me :)

**Comment by jtgeibel** [comment]
@Turbo87 I've added a test that covers some scenarios where I observed breakage. I'll probably move this test upstream eventually, but it is worth including here for now.

I've also added a command to the `crates-admin` binary, that will test existing versions and dependency requirements in the database against our current version and the latest `semver` release.

File: src/admin/test_new_semver_release.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    use self::dependencies::dsl::*;
+    use self::versions::dsl::*;
+
+    let url = db::connection_url(&dotenv::var("READ_ONLY_REPLICA_URL").unwrap());
+    let conn = PgConnection::establish(&url).unwrap();
+    conn.transaction::<_, diesel::result::Error, _>(|| {
+        let vers = versions.select(num).load(&conn)?;
+        test_versions(&vers);
+
+        let deps = dependencies.select(req).load(&conn)?;
```
Comment (jtgeibel) [inline src/admin/test_new_semver_release.rs:27]: I'm not entirely sure about loading all dependencies in a single query, but this is against the replica database and I expect memory usage to be somewhere around 100 MB so I think this is okay for our current scale.

File: src/admin/test_new_semver_release.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    println!("Test finished.");
+}
+
+fn test_versions(versions: &[String]) {
+    for version in versions {
+        if let Err(e) = semver::Version::parse(version) {
+            println!("Could not parse `{}` as a semver::Version: {}", version, e);
+        }
+    }
+
```
Comment (Turbo87) [inline src/admin/test_new_semver_release.rs:43]:
```suggestion
```

this looks like a duplicate to me, no?

File: src/admin/test_new_semver_release.rs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                version, e
+            );
+        }
+    }
+}
+
+fn test_dependency_predicates(versions: &[String]) {
+    for version in versions {
+        if let Err(e) = semver::VersionReq::parse(version) {
+            println!("Could not parse `{}` as a semver::Version: {}", version, e);
```
Comment (Turbo87) [inline src/admin/test_new_semver_release.rs:61]:
```suggestion
            println!("Could not parse `{}` as a semver::VersionReq: {}", version, e);
```

File: src/admin/test_new_semver_release.rs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        if let Err(e) = semver_next::VersionReq::parse(version) {
+            println!(
+                "Could not parse `{}` as a semver_next::Version: {}",
```
Comment (Turbo87) [inline src/admin/test_new_semver_release.rs:66]:
```suggestion
                "Could not parse `{}` as a semver_next::VersionReq: {}",
```

**Pull request closed without merging**